openbb_platform_pro_backend/main.py

```diff
@@ -99,25 +117,130 @@ async def get_widgets():
+            hub_defaults = json.loads(hub_settings.defaults.model_dump_json())
+        except Exception as e:
+            print(f"\n\nError connecting with Hub:\n{e}")
+            hub_credentials = {}
+            hub_preferences = {}
+            hub_defaults = {}
+
+        # Prompt the user to ask if they want to persist the new settings
+        persist_input = input(
+            "\n\nDo you want to persist the new settings?"
+            + " Not recommended for public machines. (yes/no): "
+        ).strip().lower()
+
+        if persist_input in ["yes", "y"]:
+            PERSIST = True
+        elif persist_input in ["no", "n"]:
+            PERSIST = False
+        else:
+            print("\n\nInvalid input. Defaulting to not persisting the new settings.")
+            PERSIST = False
+
+        # Save the current settings to restore at the end of the session.
+        if PERSIST is False:
+            with open(USER_SETTINGS_COPY, "w") as f:
+                json.dump(current_settings, f, indent=4)
+
+        new_settings = current_settings.copy()
+
+        # Update the current settings with the new settings
+        if hub_credentials:
+            for k, v in hub_credentials.items():
+                if v:
+                    new_settings["credentials"][k] = v
+
+        if hub_preferences:
+            for k, v in hub_credentials.items():
+                if v:
+                    new_settings["preferences"][k] = v
+
+        if hub_defaults:
+            for k, v in hub_defaults.items():
+                if k == "commands":
+                    for key, value in hub_defaults["commands"].items():
+                        if value:
+                            new_settings["defaults"]["commands"][key] = value
+                elif v:
+                    new_settings["defaults"][k] = v
+                else:
+                    continue
+            new_settings["defaults"].update(hub_defaults)
+
+        # Write the new settings to the user_settings.json file
+        with open(CURRENT_USER_SETTINGS, "w") as f:
+            json.dump(new_settings, f, indent=4)
+
+        current_settings = new_settings
+
+    if current_settings["credentials"].get("OPENAI_API_KEY"):
+        print("\n\nOpenAI API Key found, adding to the environment variables.\n")
+        os.environ["OPENAI_API_KEY"] = current_settings["credentials"]["OPENAI_API_KEY"]
+
     host = os.getenv("OPENBB_API_HOST", "127.0.0.1")
     if not host:
-        raise ValueError(
-            "OPENBB_API_HOST is set incorrectly. It should be an IP address or hostname."
+        print(
+            "\n\nOPENBB_API_HOST is set incorrectly. It should be an IP address or hostname."
         )
+        host = input("Enter the host IP address or hostname: ")
+        if not host:
+            host = "127.0.0.1"
 
     port = os.getenv("OPENBB_API_PORT", 8000)
+
     try:
         port = int(port)
```

Comment:
when we check the port number we actually need to check that the port number is not bellow 1024 as those are usually privileged and require sudo to run services on